sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/highlighting/XMLHighlighting.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * SonarQube XML Plugin
+ * Copyright (C) 2010 SonarSource
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.xml.highlighting;
+
+import com.google.common.io.Files;
+
+import javax.xml.stream.Location;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
+public class XMLHighlighting {
+
+  private static final String XML_DECLARATION_TAG = "<?xml";
+
+  private List<HighlightingData> highlighting = new ArrayList<>();
+  private String content;
+
+  public XMLHighlighting(File file, Charset charset) throws Exception{
+    content = Files.toString(file, charset);
+    highlightXML(new FileReader(file));
```

Comment:
@vilchik-elena Ok, almost there. FileReader always uses the default platform charset. I suggest this change:
`highlightXML(new InputStreamReader(new FileInputStream(file), charset));`